dev/src/write-batch.ts

```diff
@@ -577,7 +586,54 @@ export class WriteBatch implements firestore.WriteBatch {
       api.BatchWriteResponse
     >('batchWrite', request, tag, retryCodes);
 
-    return (response.writeResults || []).map((result, i) => {
+    let retryCount = 0;
+    while (retryCount < MAX_BATCH_WRITE_RETRY_ATTEMPTS) {
```

Comment:
`retryCount` is pretty ambiguous and doesn't clarify if the first retry loop is part of the count. I usually go with "attempt" to clarify that it is.